metrics/mobile-app-metrics.md

```diff
@@ -0,0 +1,131 @@
+
+## Proposed Solution
+
+A Metrics Service will be available from the Service Catalog, which can be provisioned by developers for use with their mobile applications.
+
+The extra components outlined in this proposal will be added to the existing components outlined in the [Metrics Endpoints and Auto Discovery Proposal](./mobile-app-metrics-service.md).
+
+The following diagram illustrates the components of the Metrics Service:
+
+![service diagram](./app-service-diagram.jpg)
+
+### Mobile App Client
+
+A native, platform-specific SDK for targetting the Metrics Service will be available to developers.
+
+Upon installation into a project and providing configuration for targetting the Metrics Service the SDK will automatically send general information about the general runtime for this application with a unique device ID, namely the `[SDK Version, clientID, timestamp]` tuple upon application initialization.
+
+The SDK will also instrument calls at the network level in order to gather metrics around communication and bandwith consumption.
+
+Other system resources utilization data can also be sent periodically.
+
+### Mobile Metrics Service
+
+A thin HTTP-based service will be present to receive metrics data from the mobile applications in an storage-agnostic way, then forwarding to the Storage service.
+
+Compared to mobile applications interfacing directly with the Storage service, this presents the following tradeoff:
+
+#### Pros
+
+- Possibility to upgrade or change the Storage service independently of mobile client SDK version and implementation;
+
+- Possible support for different Storage engines which can be selected based on application needs;
+
+#### Cons
+
+- Extra service deployment and scaling needs;
+
+- Development and maintenance of the new service;
+
+- Types of gathered metrics can become a subset of the underlying Storage's full range of types.
+
+<!-- TODO: mention logstash, fluentd -->
+
+### Storage
+
+The Storage service will store metrics data forwarded from the Mobile Metrics Service. Its implementation can be supplied by the following projects:
+
+#### Prometheus
+
+##### Pros
+
+- Shared service with backend metrics
+- Requires an additional service for long-term metrics storage
+
+##### Cons
+
+- Data model is time-series exclusive
+- Pull-based data gathering
+
+#### ElasticSearch
+
+##### Pros
+
+- High advertised performance
+- Highest variety of data types and modelling
+- Rich querying possibilities
+
+##### Cons
+
+- High minimum resource requirements (over evaluation-sized clusters)
+
+#### MongoDB
+
+<!-- TODO -->
+
+##### Pros
+
+##### Cons
+
+#### InfluxDB
+
+##### Pros
+
+- Long-term storage adapter for Prometheus
+- HTTP-based API for push-based metrics gathering
+- Flexible querying with direct comparison to SQL
+
+##### Cons
+
+- Focused on time-series data
```

Comment:
@paolobueno do you have an idea if text search is possible? I am thinking about remote logging feature we are going to provide somewhere in the future.

Reply:
ah, ok, saw it here: https://docs.influxdata.com/influxdb/v0.8/api/query_language/
`select * from events where email !~ /.*gmail.*/;`

Reply:
hmm, logs are perhaps not natural to store in InfluxDB though. we might need some additional research if Influx can be as data storage for remote logging. this is low priority for now.

Reply:
agreed on priority. Remote logging can be a separate proposal (but may or may not re-use some elements in this proposal).
There are no high level stories/trellos for a Remote Logging service at this time (but good to be aware of it long term)